clang/lib/Analysis/LifetimePsetBuilder.cpp

```diff
@@ -778,19 +776,32 @@ class PSetsBuilder : public ConstStmtVisitor<PSetsBuilder, void> {
 
   bool CheckPSetValidity(const PSet &PS, SourceRange Range);
 
-  /// Invalidates all psets that point to V or something owned by V
-  void invalidateVar(Variable V, unsigned Order, InvalidationReason Reason) {
+  void invalidateVar(Variable V, InvalidationReason Reason) {
+    for (auto &I : PMap) {
+      const auto &Var = I.first;
+      PSet &PS = I.second;
+      if (PS.containsInvalid())
+        continue; // Nothing to invalidate
+
+      if (PS.containsBase(V)) {
```

Comment:
You can elide the braces if the if-body is a single line

Reply:
Done